metals/src/main/scala/scala/meta/internal/metals/QuickFixes.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta.internal.metals
+
+import scala.concurrent.Future
+import scala.meta.pc.CancelToken
+import org.eclipse.{lsp4j => l}
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.metals.MetalsEnrichments._
+
+trait QuickFixes {
+  trait QuickFix {
+    def contribute(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]]
+  }
+
+  object QuickFix {
+
+    def fromDiagnostics(
+        diagnostics: Seq[l.Diagnostic],
+        params: l.CodeActionParams,
+        compilers: Compilers,
+        token: CancelToken
+    )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] =
+      Future
+        .sequence(
+          diagnostics
+            .collect {
+              case d @ ScalacDiagnostic.SymbolNotFound(name) =>
+                ImportMissingSymbol(
+                  name,
+                  d,
+                  params,
+                  compilers,
+                  token
+                )
+            }
+            .map(_.contribute)
+        )
+        .map(_.flatten)
+  }
+
+  final case class ImportMissingSymbol(
+      name: String,
+      diagnostic: l.Diagnostic,
+      params: l.CodeActionParams,
+      compilers: Compilers,
+      token: CancelToken
+  ) extends QuickFix {
+
+    override def contribute(
+        implicit ec: ExecutionContext
+    ): Future[Seq[l.CodeAction]] = {
+
+      // TODO(gabro): this is hack. Instead of computing the auto-imports for a name at a range,
```

Comment:
To replace this hack we would need most likely to add an additional method to the java interfaces.

Reply:
Yep, I’ve actually started on that, but then decide to get some early feedback first!